sdk/eventhub/eventhubs-checkpointstore-table/src/tableCheckpointStore.ts

```diff
@@ -42,13 +109,69 @@ export class TableCheckpointStore implements CheckpointStore {
    *  - `tracingOptions`: Options for configuring tracing.
    * @returns A list partitions this instance successfully claimed ownership.
    */
-  async claimOwnership(): /*
-    partitionOwnership: PartitionOwnership[],
-    options: OperationOptions = {}
-    */
-  Promise<PartitionOwnership[]> {
-    console.log("nothing to claim");
-    throw new Error("not implemented");
+  async claimOwnership(partitionOwnership: PartitionOwnership[]): Promise<PartitionOwnership[]> {
+    const partitionOwnershipArray: PartitionOwnership[] = [];
+
+    for (const ownership of partitionOwnership) {
+      const partition_Key = `${ownership.fullyQualifiedNamespace} ${ownership.eventHubName} ${ownership.consumerGroup} Ownership`;
+      const ownershipEntity: PartitionOwnershipEntity = {
+        partitionKey: partition_Key,
+        rowKey: ownership.partitionId,
+        ownerid: ownership.ownerId
+      };
+
+      // When we have an etag, we know the entity existed.
+      // If we encounter an error we should fail.
+      try {
+        if (ownership.etag) {
+          const updatedMetadata = await this._tableClient.updateEntity(ownershipEntity, "Replace", {
+            etag: ownership.etag
+          });
+          ownership.etag = updatedMetadata.etag;
+          ownership.lastModifiedTimeInMs = await this._toMillisecs(
```

Comment:
So it looks like we have the same bug that python did - we're mutating the list of ownership the user passes in rather than returning completely new ownership items. This is a bug in our storage-blob implementation as well.

We don't want to mutate the ownership that the user passes in, so we should create a copy of the ownership first, then mutate the copy. It's the copy we'll push to the `partitionOwnershipArray`.

We're fortunate that we don't have any nested properties, so to copy an ownership is fairly straightforward, something like this:
```ts
const updatedOwnership = {...ownership};
```

`{...ownership}` creates a new object, and copies every field from `ownership` into the new object. Since we're only dealing with primitive types (strings, numbers) this works fine for our needs.

Reply:
> ownership.lastModifiedTimeInMs = await this._toMillisecs((updatedMetadata as any).Timestamp

Is Timestamp actually defined when you call updateEntity? I logged https://github.com/Azure/azure-sdk-for-js/issues/16733 because I wasn't seeing a way to get it. I think for now you're going to be stuck getting the timestamp via a `getEntity` call or a `queryEntities` call.